hax/treemakers/posrec.py

```diff
@@ -0,0 +1,163 @@
+import hax
+import numpy as np
+from hax.minitrees import TreeMaker
+from pax.PatternFitter import PatternFitter
+from pax.configuration import load_configuration
+from pax.InterpolatingMap import InterpolatingMap
+from pax import utils
+from pax import exceptions
+from scipy.stats import binom_test
+import json
+from keras.models import model_from_json, Sequential
+from keras.layers import Dense, Dropout, Activation, Reshape, Flatten
+from keras import regularizers
+
+class PositionReconstruction(TreeMaker):
+    """Stores position-reconstruction-related variables.
+    
+    Provides:
+       - s1_pattern_fit: s1 pattern fit computed with corrected
+                         position and areas
+       - s1_pattern_fit_hits: s1 pattern fit computed with corrected
+                         position and hits
+
+    """
+    __version__ = '0.12'
+    extra_branches = ['peaks.area_per_channel[260]',
+                      'peaks.hits_per_channel[260]',
+                      'peaks.n_saturated_per_channel',
+                      'interactions.x','interactions.y','interactions.z']
+
+    def __init__(self):
+        
+        hax.minitrees.TreeMaker.__init__(self)
+        
+        # We need to pull some stuff from the pax config
+        self.pax_config = load_configuration("XENON1T")
+        self.tpc_channels = list(range(0,   247+1))
+        self.confused_s1_channels = []
+        self.statistic = self.pax_config \
+                         ['BuildInteractions.BasicInteractionProperties'] \
+                         ['s1_pattern_statistic']
+        qes = np.array(self.pax_config['DEFAULT']['quantum_efficiencies'])
+
+        self.pattern_fitter = PatternFitter(
+            filename=utils.data_file_name(
+                self.pax_config['WaveformSimulator']['s1_patterns_file']),
+            zoom_factor = self.pax_config['WaveformSimulator'].get(
+                's1_patterns_zoom_factor', 1),
+            adjust_to_qe=qes[self.tpc_channels],
+            default_errors=(self.pax_config['DEFAULT']['relative_qe_error'] +
+                            self.pax_config['DEFAULT']['relative_gain_error'])
+        )
+
+        # For s1_aft variable
+        # Don't yell at me for hardcoding the filename into hax because it was
+        # also hardcoded in pax. Just kicking the can.
+        #aftmap_filename = utils.data_file_name('s1_aft_xyz_XENON1T_06Mar2017.json')
+        aftmap_filename = utils.data_file_name('s1_aft_xyz_XENON1T_20170808.json')
```

Comment:
move to ```hax.ini``` like other maps in ```corrections.py```?

Reply:
Fixed in https://github.com/XENON1T/hax/pull/174/commits/ef6fd7de164d451d2e3ee127b35765c72246b28b